assets/js/components/notifications/dashboard-auth-alert.js

```diff
@@ -24,26 +29,95 @@ import { __, sprintf } from '@wordpress/i18n';
+import { getModulesData } from '../../util';
+import { STORE_NAME as CORE_USER } from '../../googlesitekit/datastore/user/constants';
+const { useSelect } = Data;
+
+// Map of scope IDs to Site Kit module slugs.
+const scopeIDToSlug = {
+	siteverification: 'site-verification',
+	webmasters: 'search-console',
+};
+const MESSAGE_MULTIPLE = 'multiple';
+const MESSAGE_SINGULAR = 'single';
+const MESSAGE_GENERIC = 'generic';
+
+function mapScopesToModuleNames( scopes ) {
+	const modules = getModulesData();
+
+	return scopes
+		// Map into an array of matches.
+		.map( ( scope ) => scope.match( /^https:\/\/www.googleapis.com\/auth\/([a-z]+)/ ) )
+		// Map each match into a module slug, if any.
+		.map( ( [ , id ] ) => scopeIDToSlug[ id ] || id )
+		// Map module slugs into module names. If there is no matched module, set to `false`.
+		.map( ( slug ) => modules[ slug ]?.name || false )
+	;
+}
 
 const DashboardAuthAlert = () => {
-	const { admin: { connectURL } } = global.googlesitekit;
-	const { currentAdminPage } = global.googlesitekit.admin;
-	const product = currentAdminPage
-		.replace( /googlesitekit|module|-/g, ' ' )
-		.replace( /(^\w{1})|(\s{1}\w{1})/g, ( match ) => match.toUpperCase() )
-		.trim();
+	const unsatisfiedScopes = useSelect( ( select ) => select( CORE_USER ).getUnsatisfiedScopes() );
+
+	// Selector for `getConnectURL` not merged yet.
+	// const connectURL = useSelect( ( select ) => select( CORE_USER ).getConnectURL() );
+	// TODO: replace use of global with above selector once merged.
+	const connectURL = global.googlesitekit?.admin?.connectURL;
+
+	if ( unsatisfiedScopes === undefined || connectURL === undefined ) {
+		return null;
+	}
+
+	let messageID;
+	let moduleNames;
+	// Determine if all scopes are in Google API format, otherwise use generic message.
+	if ( unsatisfiedScopes.some( ( scope ) => ! scope.match( /^https:\/\/www.googleapis.com\/auth\// ) ) ) {
+		messageID = MESSAGE_GENERIC;
+	} else {
+		// All scopes are in Google API format, map them to module names.
+		moduleNames = mapScopesToModuleNames( unsatisfiedScopes );
+		// If any scope did not resolve to a module name, use the generic message.
+		if ( moduleNames.some( ( name ) => name === false ) ) {
+			messageID = MESSAGE_GENERIC;
+		} else {
+			moduleNames = unique( moduleNames );
+			messageID = 1 < moduleNames.length ? MESSAGE_MULTIPLE : MESSAGE_SINGULAR;
+		}
+	}
+
+	let message;
+
+	switch ( messageID ) {
+		case MESSAGE_MULTIPLE:
+			/* translators: used between list items, there is a space after the comma. */
+			const listSeparator = __( ', ', 'google-site-kit' );
+			message = sprintf(
+				/* translators: %s: List of product names */
+				__( 'Site Kit can’t access the some relevant data because you haven’t granted all permissions requested during setup. To use Site Kit, you’ll need to redo the setup for: %s – make sure to approve all permissions at the authentication stage.', 'google-site-kit' ),
```

Comment:
```suggestion
				__( 'Site Kit can’t access all relevant data because you haven’t granted all permissions requested during setup. To use Site Kit, you’ll need to redo the setup for: %s – make sure to approve all permissions at the authentication stage.', 'google-site-kit' ),
```